### PR TITLE
feat(dev-server): Add toast for successful sync and warning for failure

### PR DESCRIPTION
**Pull request by cdelst**
image

### CODE REVIEW

File: internal/dev_server/ui/src/main.tsx [hunk 1]
```diff
@@ -28,6 +29,7 @@ const Root = () => {
     <React.StrictMode>
       <IconProvider>
         <App />
```
Comment (mike-zorn) [inline internal/dev_server/ui/src/main.tsx:31]: Should it be at the top? That's where most of the latent interactions happen.

**Review by mike-zorn — APPROVED** [review]
One thought on placement, but 👍🏻

File: internal/dev_server/ui/src/main.tsx (hunk 1 above)
Comment (cdelst) [reply internal/dev_server/ui/src/main.tsx:31]: TBH, not sure.  I think just having it somewhere relatively global is standard.

Don't think it matters that much here?

Comment (cdelst) [reply internal/dev_server/ui/src/main.tsx:31]: wym by at the top?

Comment (mike-zorn) [reply internal/dev_server/ui/src/main.tsx:31]: oh like at the top of the page. Screenshot makes it look like maybe it's at the bottom of the page?

Comment (mike-zorn) [reply internal/dev_server/ui/src/main.tsx:31]: Location in the markup seems fine.

Comment (cdelst) [reply internal/dev_server/ui/src/main.tsx:31]: ohhh, yea... bottom of the page is standard for toasts and I don't think We can change that.  The <ToastProvider> doesn't actually control that either.  The position is absolute no matter where we render the provider

Comment (cdelst) [reply internal/dev_server/ui/src/main.tsx:31]: When using it it's clear enough to give some confirmation 👍🏻

Comment (mike-zorn) [reply internal/dev_server/ui/src/main.tsx:31]: 🆒

**Pull request merged**